src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -247,20 +272,64 @@ public void RemoveItems(IEnumerable collection)
         /// <returns>IDisposable.</returns>
         public IDisposable SuspendChangeNotifications()
         {
-            return new DisposableToken<FastObservableCollection<T>>(this,
+            return SuspendChangeNotifications(SuspensionMode.None);
+        }
+
+        /// <summary>
+        /// Suspends the change notifications until the returned <see cref="IDisposable"/> is disposed.
+        /// <example>
+        /// <code>
+        /// <![CDATA[
+        /// var fastCollection = new FastObservableCollection<int>();
+        /// using (fastCollection.SuspendChangeNotificaftions())
+        /// {
+        ///     // Adding or removing events inside here will not raise events
+        ///     fastCollection.Add(1);
+        ///     fastCollection.Add(2);
+        ///     fastCollection.Add(3);
+        /// 
+        ///     fastCollection.Remove(3);
+        ///     fastCollection.Remove(2);
+        ///     fastCollection.Remove(1);
+        /// }
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="mode">The suspension Mode.</param>
+        /// <returns>IDisposable.</returns>
+        public IDisposable SuspendChangeNotifications(SuspensionMode mode)
+        {
+            if (_suspensionContext == null)
+            {
+                // Create new context
+                _suspensionContext = new SuspensionContext<T>(mode);
+            }
+            else if (_suspensionContext != null && _suspensionContext.Mode != mode)
+            {
+                throw Log.ErrorAndCreateException<InvalidOperationException>("Cannot change mode during another active suspension.");
```

Comment:
Nice one, good usage of logging and exception throwing at the same time as well.